.NET/Microsoft.Recognizers.Text.Number/Extractors/BaseNumberExtractor.cs

```diff
@@ -64,6 +66,20 @@ public virtual List<ExtractResult> Extract(string source)
                         continue;
                     }
 
+                    // Matches containing separators 'in', 'out of' should be considered fractions only when numerator < denominator
+                    if (m.Groups["ambiguousSeparator"].Success)
+                    {
+                        var numerator = m.Groups["numerator"];
+                        var denominator = m.Groups["denominator"];
+                        int num = ParseNumber(numerator);

```

Comment:
This can be very costly. We need to run some perf impact tests here.

Reply:
yes I was worried about it, especially the use of NumberParser. But since it runs only when ambiguousSeparator is matched, it should have an impact only for the cases "N in M" or "N out of M"